### PR TITLE
[Bagel] Support Cache-Dit

### PR DESCRIPTION
**Pull request by princepride**
## Purpose

This PR enables `cache-dit` acceleration for the **Bagel** model in `vllm-omni`.

The integration presents a unique challenge: Bagel leverages a `NaiveCache` object passed via `past_key_values` for its KV-cache mechanism, whereas `cache-dit`'s standard `ForwardPattern` expects an `encoder_hidden_states` tensor (typically used for cross-attention in DiTs) and attempts to perform tensor operations (like residual calculation) on it.

To resolve this **without modifying the original `cache-dit` library or the Bagel model definition**, this PR implements a set of custom adapter classes that inherit from and extend `cache-dit`'s core components:

1.  **`BagelCachedContextManager`**: Inherits from `CachedContextManager`. It overrides `apply_cache` to add `isinstance(..., torch.Tensor)` checks. This ensures that when `NaiveCache` objects are passed through the pipeline (masquerading as `encoder_hidden_states` in the pattern), they are preserved as-is without crashing due to invalid tensor operations (like `.contiguous()` or subtraction).

2.  **`BagelCachedBlocks`**: Inherits from `CachedBlocks_Pattern_0_1_2`.
    *   **Signature Mapping**: It handles the argument mapping between `cache-dit`'s expected `encoder_hidden_states` and Bagel's `past_key_values`. The `forward` method intercepts `past_key_values` from kwargs, and internal block call methods (`call_Fn/Mn/Bn_blocks`) ensure the underlying Bagel decoder layer receives arguments in the correct keyword format preventing `TypeError`.
    *   **Safe Residuals**: Overrides `call_Mn_blocks` and `compute_or_prune` to skip residual calculation for non-tensor `past_key_values`, protecting the `NaiveCache`.

3.  **`BagelCachedAdapter`**: Inherits from `CachedAdapter`. It serves as the entry point, orchestrating the instantiation of the custom `BagelCachedContextManager` and `BagelCachedBlocks` instead of the default ones.

These adapters are defined in `vllm_omni/diffusion/cache/cache_dit_backend.py` and applied specifically for Bagel pipelines.

## Test Plan

**Test Command:**
```
import torch
import time
from vllm_omni.entrypoints.omni_diffusion import OmniDiffusion

def run_test(cache_backend=None, cache_config=None, run_name="Baseline"):
    pipeline = OmniDiffusion(
        model="../models/BAGEL-7B-MoT",
        cache_backend=cache_backend,
        cache_config=cache_config
    )
    prompt = "A woman wear a blue dress"
    start_gen = time.time()
    result = pipeline.generate(
        prompt, 
        seed=52
    )
    end_gen = time.time()
    gen_time = end_gen - start_gen
    print(f"Generation Time: {gen_time:.4f}s")
    
    output_path = f"output_bagel_{run_name.lower().replace(' ', '_')}.png"
    result.images[0].save(output_path)

def main():
    # 1. Baseline (No Cache)
    baseline_time = run_test(cache_backend=None, cache_config=None, run_name="Baseline (No Cache)")
    
    cache_config = {
        # DBCache parameters [cache-dit only]
        "Fn_compute_blocks": 1,  # Optimized for single-transformer models
        "Bn_compute_blocks": 0,  # Number of backward compute blocks
        "max_warmup_steps": 4,  # Maximum warmup steps (works for few-step models)
        "residual_diff_threshold": 0.24,  # Higher threshold for more aggressive caching
        "max_continuous_cached_steps": 3,  # Limit to prevent precision degradation
        # TaylorSeer parameters [cache-dit only]
        "enable_taylorseer": False,  # Disabled by default (not suitable for few-step models)
        "taylorseer_order": 1,  # TaylorSeer polynomial order
        # SCM (Step Computation Masking) parameters [cache-dit only]
        "scm_steps_mask_policy": None,  # SCM mask policy: None (disabled), "slow", "medium", "fast", "ultra"
        "scm_steps_policy": "dynamic",  # SCM steps policy: "dynamic" or "static"
    }
    run_test(cache_backend="cache_dit", cache_config=cache_config, run_name="Cache-Dit")

if __name__ == "__main__":
    main()

```
[Optional] You can directly use `examples/offline_inference/text_to_image/text_to_image.py`
```bash
python examples/offline_inference/text_to_image/text_to_image.py --model ../models/BAGEL-7B-MoT --cache_backend cache_dit
```

## Test Result

**Performance Comparison:**

| Metric | Baseline (No Cache) | Cache-Dit Enabled | Speedup |
| :--- | :--- | :--- | :--- |
| **Generation Time** | 9.28s | 4.41s | **~2.1x** |

**Visual Validation:**

<table style="width: 100%; border-collapse: collapse;">
  <tr>
    <td align="center" style="border: none;">
      <br/>
      <b>Baseline (No Cache)</b>
    </td>
    <td align="center" style="border: none;">
      <br/>
      <b>Cache-Dit Enabled</b>
    </td>
  </tr>
</table>

### CODE REVIEW

File: vllm_omni/diffusion/cache/cache_dit_backend.py [hunk 1]
```diff
@@ -401,6 +409,384 @@ def refresh_cache_context(pipeline: Any, num_inference_steps: int, verbose: bool
...
+        torch._dynamo.graph_break()
+        if can_use_prune:
+            self.context_manager.add_pruned_step()
+            hidden_states, encoder_hidden_states = self.context_manager.apply_prune(
+                hidden_states,
+                encoder_hidden_states,
+                prefix=(
+                    f"{self.cache_prefix}_{block_id}_Bn_residual"
+                    if self.context_manager.is_cache_residual()
+                    else f"{self.cache_prefix}_Bn_hidden_states"
```
[the remark this answers is not in the document]
Comment (princepride) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:543]: I directly copied from cache-dit

File: vllm_omni/diffusion/cache/cache_dit_backend.py [hunk 2]
```diff
@@ -401,6 +409,384 @@ def refresh_cache_context(pipeline: Any, num_inference_steps: int, verbose: bool
     return refresh_cache_context
 
 
+class BagelCachedContextManager(CachedContextManager):
```
Comment (SamitHuang) [inline vllm_omni/diffusion/cache/cache_dit_backend.py:412]: Do you think it's feasible to add it directly in cache-dit, so that cache-dit repo can support bagel natively as well? @DefTruth

Comment (DefTruth) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: That sounds good to me. Feel free to send a PR to cache-dit, and I can help review it.

Comment (hsliuustc0106) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: @princepride PTAL

Comment (princepride) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: Bagel's original code is extremely complex, and it doesn't name its modules according to the standard Dit model. I can only guarantee that the code I submit will support Bagel's acceleration on vLLM-Omni.

Comment (princepride) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: BTW, I believe hunyuan image will also benefit from this new feature.

Comment (DefTruth) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: May I ask if Diffusers supports Bagel? If not, it seems that supporting it in cache-dit doesn't make much sense either.

Comment (DefTruth) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: I think it would be best to first support Bagel cache acceleration in vllm-omni

Comment (princepride) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: Sorry, how do I know if Diffusers supports them? They all use their own code execution models and employ a unified kv cache full-modal LLM architecture.

Comment (DefTruth) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: It seems that Diffusers does not support Bagel yet. I can't found the Bagel-related code from its code.

Comment (princepride) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: The same as Hunyuan Image 3.0.

Comment (princepride) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: @SamitHuang @hsliuustc0106 What do you think?

Comment (SamitHuang) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: indeed, diffusers doesn't support bagel or HYI3.0. CacheDiT doesn't plan to support image/video generation models in transformers, right? @DefTruth

Comment (DefTruth) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: since cache-dit is built on top of the diffusers library, we have long focused mainly on the models supported in diffusers.

Comment (hsliuustc0106) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: so the conclusion is still keeping the change in vllm-omni here?

**Comment by hsliuustc0106** [comment]
please update the diffusion acceleration md for this support

File: vllm_omni/diffusion/cache/cache_dit_backend.py (hunk 2 above)
Comment (princepride) [reply vllm_omni/diffusion/cache/cache_dit_backend.py:412]: > so the conclusion is still keeping the change in vllm-omni here?

I'm afraid so.

**Comment by princepride** [comment]
@hsliuustc0106 Already update docs.

**Comment by princepride** [comment]
@tjtanaa amd CI failed, PTAL.
